app/controllers/user_notifications_controller.rb

```diff
@@ -21,4 +21,19 @@ def delete
     ).destroy!
     render nothing: true
   end
+
+  def unsubscribe_from_all_notifications
+    user_notification = UserNotification.find_by(token: params[:token])
+    user_notifications = UserNotification.where(user_id: user_notification.user_id)
+    user_notifications.each do |user_notification|
+      user_notification.destroy! if user_notification.present?
```

Comment:
Don't think the `user_notification.present?` check is still necessary. `UserNotification.where(...)` are all present.

Reply:
i think there is some kind of moment when people input token that not exist. when it happen, ```user_notifications``` also become nil class. therefore, nil.```destroy!``` will throw an error which is better to be avoided

Reply:
Won't that be `user_notification` that is `nil` instead of `user_notifications`?

Reply:
yes, ```user_notification``` is ```nil``` which makes ```user_notification.user_id``` also ```nil``` lead ```user_notifications``` to become ```nil``` too